src/devices/src/pseudo/persist.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the structures needed for saving/restoring block devices.
+
+use std::io;
+
+use snapshot::Persist;
+use utils::time::TimestampUs;
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_memory::GuestMemoryMmap;
+
+use super::*;
+
+// No value in persisting boot timer information
+// This module restores a fresh BootTimer device.
+#[derive(Versionize)]
+pub struct BootTimerState {
+    start_ts: String,
+}
+
+pub struct BootTimerConstructorArgs {
+    pub mem: GuestMemoryMmap,
+}
+
+impl Persist<'_> for BootTimer {
```

Comment:
I think there's no point in actually implementing `Persist` for `BootTimer`. @sandreim what do you think? how can we most elegantly handle a device with no state?

I need to check device-manager `Persist` impl, but I'm guessing that's where the trick will be. For this type of device don't serialize any device specific data, and on restore path just build `new()` one.

Reply:
I've removed `Persist` for `BootTimer` and had a stab at recreating a new device on restore.  It required a change to the restored device manager from using `dummy_mmio_base` to the actual `MMIO_MEM_START` address,  so that the timer get recreated with the correct address.  It would also assume that the `BootTimer` is always the first device on the bus,  which it is currently for `x86_64` but isn't for `aarch64`.   Not sure if that approach works,  but obviously happy to go with whatever you advise.....